### PR TITLE
DEBT-1419 Upgrade chips-rest-interfaces-consumer log4j to safe version

### PR DESCRIPTION
**Pull request by mattch1**
As well as the structured logging version update added the temporary override import ref to log4j 2.15 and updated jib.

### CODE REVIEW

**Review by ljgeorge — APPROVED** [review]

**Review by ljgeorge — CHANGES_REQUESTED** [review]
You need to check your concourse  build

**Comment by mattch1** [comment]
Have checked it and am upgrading those dependencies quoted in the security report.

**Comment by mattch1** [comment]
For some reason I have had to explicitly request version 3.0.0 of kafka-clients. Even though spring-kafka is supposed to pull down this version, it will persistently get version 2.7.1 which had the vulnerabilities.

**Review by JMetherell — DISMISSED** [review]

File: pom.xml [hunk 1]
```diff
@@ -85,6 +86,23 @@
             <artifactId>rest-service-common-library</artifactId>
             <version>${rest-service-common-library.version}</version>
         </dependency>
+        <!--  remove these 3 log4j once spring boot upgrade is available after 23/12/2021 -->
+        <dependency>
+            <groupId>org.apache.logging.log4j</groupId>
+            <artifactId>log4j-core</artifactId>
+            <version>2.15.0</version>
```
Comment (markpit) [inline pom.xml]: will need to upgrade to 2.16.0 as apparently 2.15.0 has a security flaw

Comment (mattch1) [reply pom.xml]: Upgraded.

File: pom.xml [hunk 2]
```diff
@@ -43,6 +44,14 @@
 
     <dependencyManagement>
         <dependencies>
+            <!--  remove these 3 log4j once spring boot upgrade is available after 23/12/2021 -->
+            <dependency>
+                <groupId>org.apache.logging.log4j</groupId>
+                <artifactId>log4j-bom</artifactId>
+                <version>2.16.0</version>
+                <scope>impo rt</scope>
```
Comment (markpit) [inline pom.xml]: there is a type-o in <scope>import which is causing it to not bring in 2.16.0

**Review by markpit — DISMISSED** [review]

File: pom.xml [hunk 3]
```diff
@@ -43,6 +44,14 @@
 
     <dependencyManagement>
         <dependencies>
+            <!--  remove these 3 log4j once spring boot upgrade is available after 23/12/2021 -->
```
Comment (sroberts2) [inline pom.xml]: You should no longer need this block if you use version 1.9.11 of structured logging

**Review by sroberts2 — CHANGES_REQUESTED** [review]

File: pom.xml (hunk 3 above)
Comment (mattch1) [reply pom.xml]: Removed the comment but if I take out the dependecy block the dependecny reverts to 2.14.0 which is unsafe and will result in the security checks failing this is  in spite of having updated structured logging to 1.9.11

Comment (mattch1) [reply pom.xml]: Removed the block now but we are getting security complaint from concourse.

**Review by sroberts2 — DISMISSED** [review]

**Review by markpit — DISMISSED** [review]

**Comment by ljgeorge** [comment]
The build is failing because a older version of log4j appears to be in the product. I suggest asking devops if the workspace can be cleared down and if you don't get an answer in the next 2 hours you go back to fixing. the version of log4j.

**Review by ljgeorge — APPROVED** [review]

**Review by sroberts2 — APPROVED** [review]

**Pull request merged**